modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutableTableRegistryImpl.java

```diff
@@ -77,6 +83,22 @@ public void onSchemaUpdated() {
         tableCache.clear();
     }
 
+    private CompletableFuture<ExecutableTable> loadTable(String tableName, TableDescriptor tableDescriptor) {
+        return tableManager.tableAsyncInternal(tableName.toUpperCase())
+                .thenApply(table -> {
+                    InternalTable internalTable = table.internalTable();
+                    SchemaRegistry schemaRegistry = schemaManager.schemaRegistry(table.tableId());
+                    SchemaDescriptor schemaDescriptor = schemaRegistry.schema();
+                    TableRowConverter rowConverter = new TableRowConverterImpl(schemaRegistry, schemaDescriptor, tableDescriptor);
+                    ScannableTable scannableTable = new ScannableTableImpl(internalTable, rowConverter, tableDescriptor);
+
+                    UpdatableTableImpl updatableTable = new UpdatableTableImpl(table.tableId(), tableDescriptor, internalTable.partitions(),
+                            replicaService, clock, rowConverter, schemaDescriptor);
+
+                    return new ExecutableTableImpl(internalTable, scannableTable, updatableTable);
```

Comment:
why we assembly  ExecutableTableImpl from 3 components and further use in ro and rw methods ? 
```
public IgniteRel visit(IgniteTableScan rel) {
...
public IgniteRel visit(IgniteTableModify rel) {
...
resolveTable(igniteTable);
```
looks strange a bit, did it need to be reworked ? I just think that UpdatableTableImpl is not need for IgniteTableScan, isn`t it ?

Reply:
Agree, I think `UpdatableTableImpl` can be instantiated on-demand.
I won't do any unnecessary refactoring in this PR, or in feature branch.

Reply:
Let's create a ticket for this.

Reply:
Done.
https://issues.apache.org/jira/browse/IGNITE-20231